deps/gpg-error.BUILD.bazel

```diff
@@ -0,0 +1,267 @@
+    srcs = ["src/gen-posix-lock-obj.c", "src/posix-lock-obj.h", "src/config.h"],
+    local_defines = ["HAVE_CONFIG_H"],
+    includes = ["src"],
+)
+
+genrule(
+    name = "lock-obj-pub_native_h",
+    outs = ["lock-obj-pub.native.h"],
+    tools = [":gen-posix-lock-obj"],
+    cmd = "$(location :gen-posix-lock-obj) > $@",
+)
+
+# The mkheader tool will use the folder containing gpg-error.h.in as the source
+# folder contanining all the other includes, except for the lock object one which *must*
+# be found in the CWD.
+# config.h is given through the command line so it doesn't need to be moved somewhere special.
+# - triplet (see mkheader.c for the list of valid values)
+# - path to gpg-error.h.in template
+# - path to config.h
+# - the version in usual (ie. major.minor) format
+# - the version as hexadecimal number (0x{major}{minor}00)
+mkheader_cmds = [
+    "cp $(location :lock-obj-pub_native_h) $$(pwd) && $(location :mkheader) {} $(location src/gpg-error.h.in) $(location src/config.h) 1.47 0x012F00 > $@".format(arch)
+for arch in ["x86_64-unknown-linux-gnu", "arm-unknown-linux-gnueabi"] ]
+
+genrule(
+    name = "gpg-error_h",
+    srcs = [
+        "src/config.h",
+        "src/err-codes.h.in",
+        "src/err-sources.h.in",
+        "src/errnos.in",
+        "src/gpg-error.h.in",
+        ":lock-obj-pub_native_h",
+    ],
+    tools = [":mkheader"],
+    outs = ["src/gpg-error.h"],
+    cmd = select({
+        "@//:linux_x86_64": mkheader_cmds[0],
+        "@//:linux_arm64": mkheader_cmds[1],
+    })
+)
+
+cc_library(
+    name = "libgpg-error",
+    srcs = [
+        "src/posix-lock.c",
+        "src/posix-lock-obj.h",
+        "src/posix-thread.c",
+        "src/spawn-posix.c",
+        "src/gpgrt-int.h",
+        "src/protos.h",
+        "src/init.c",
+        "src/init.h",
+        "src/version.c",
+        "src/lock.h",
+        "src/thread.h",
+        "src/estream.c",
+        "src/estream-printf.c",
+        "src/estream-printf.h",
+        "src/strsource.c",
+        "src/strerror.c",
+        "src/code-to-errno.c",
+        "src/code-from-errno.c",
+        "src/visibility.c",
+        "src/visibility.h",
+        "src/sysutils.c",
+        "src/stringutils.c",
+        "src/strlist.c",
+        "src/name-value.c",
+        "src/syscall-clamp.c",
+        "src/logging.c",
+        "src/b64dec.c",
+        "src/b64enc.c",
+        "src/argparse.c",
+        "src/gettext.h",
+        # Generated sources
+        "src/config.h",
+        ":code_to_errno_h",
+        ":code_from_errno_h",
+        ":err-codes_h",
+        ":err-sources_h",
+    ],
+    hdrs = [
+        ":gpg-error_h",
+    ],
+
+    local_defines = [
+        "GPG_ERR_ENABLE_GETTEXT_MACROS=1",
+        "GPGRT_ENABLE_ES_MACROS=1",
+        "GPGRT_ENABLE_LOG_MACROS=1",
+        "GPGRT_ENABLE_ARGPARSE_MACROS=1",
+        "HAVE_CONFIG_H",
+        "LOCALEDIR=NULL",
+    ],
+    copts = [
+        "-isystem external/+_repo_rules2+gpg-error/src",
```

Comment:
This is really brittle though.  A future bazel update will change rules2 back to rules, and break all our code.
I think includes=[src] may be what you need.

Reply:
> This is really brittle though

It definitely is, and this isn't mergeable as such IMO

> I think includes=[src] may be what you need.

That works for `gpg-error` but fails when building `gcrypt` as it would include `gpg-error`'s configure instead of `gcrypt`'s

We need to provide the folder containing `config.h` through `-isystem` since it is included with `#include <config.h>`, so either with some `copts` or with `includes`, but using include would make libgpg-error's config.h visible to all the targets that depend on it which is undesirable

Reply:
@chouquette I've just added a minimally intrusive fix in commit 1e41da0510795bc4b4ec52858710755b57c2ec41.

Reply:
Thanks! I'm not sure I understand how this changes the behavior though, but I can't argue that it does

Is this a matter of ordering with the added `includes` directive taking precedence in `gcrypt`?